src/components/sbb-dialog/sbb-dialog.tsx

```diff
@@ -343,7 +325,7 @@ export class SbbDialog implements ComponentInterface, AccessibilityProperties {
       <sbb-button
         class="sbb-dialog__close"
         accessibility-label={this.accessibilityCloseLabel || i18nCloseDialog[this._currentLanguage]}
-        accessibility-controls={this.dialogId}
+        accessibility-controls={this._dialogId}
```

Comment:
this will not work, but you can try the following and check with screenreader
```suggestion
        aria-controls={this._dialogId}
```